subsys/debug/coredump/coredump_backend_mem_window.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/debug/coredump.h>
+#include "coredump_internal.h"
+
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+LOG_MODULE_REGISTER(coredump, CONFIG_KERNEL_LOG_LEVEL);
+
+static int error;
+
+static void coredump_mem_window_backend_start(void)
+{
+	/* Reset error */
+	error = 0;
+
+	while (LOG_PROCESS()) {
+		;
+	}
+
+	LOG_PANIC();
+	LOG_ERR(COREDUMP_PREFIX_STR COREDUMP_BEGIN_STR);
+}
+
+static void coredump_mem_window_backend_end(void)
+{
+	if (error != 0) {
+		LOG_ERR(COREDUMP_PREFIX_STR COREDUMP_ERROR_STR);
+	}
+
+	LOG_ERR(COREDUMP_PREFIX_STR COREDUMP_END_STR);
+}
+
+static void coredump_mem_window_backend_buffer_output(uint8_t *buf, size_t buflen)
+{
+	uint8_t *mem_window_sink = (uint8_t *)(ADSP_DW->slots[1]);
```

Comment:
Hmm, shouldn't this be declared in some included header file? I fail to see where ADSP_DW is declared.

Reply:
@kv2019i , mtrace set desc.type to MTRACE_LOGGING_SLOT_TYPE, we need to set desc[1] to ADSP_DW_SLOT_TELEMETRY here and kernel driver will check desc type to find this slot.
````c
static void mtrace_init(void)
{
	ADSP_DW->descs[0].type = MTRACE_LOGGING_SLOT_TYPE(MTRACE_CORE);
}
````

Reply:
@RanderWang What does it mean for me? Do I have to change it to desc[0] ?

Reply:
Ack @RanderWang , so we need to agree how host should identify the coredump slot. Out of these, coredump has been one entry of the telemetry slot, so that's probably best starting point. Not sure should if we should use some distinct layout to allow host to differentiate whether FW is following the older layout of (like https://github.com/zephyrproject-rtos/zephyr/pull/53785/files#diff-1aefb5a090e5a4c14dd8166d1a3a3067df5e1e93f5e2600324048a12494d25bfR151 with separators) or this PR. We could have perhaps a different 32bit separator byte at the start of the telemetry slot to indicate which variant is used. This doesn't have to be anything too fancy, but it will help if host has some means to identify the layout.

Reply:
I think that finally coredump will be as a part of telemetry stucture. As I mention here https://github.com/zephyrproject-rtos/zephyr/pull/53785/files#r1176417306 that structure for zephyr is outdated if we are talking about coredump. In my opinion last field of telemetry should be pointer and data length to coredump and that's it.

Reply:
@RanderWang I've made changes. Please check it out.